### DISCUSSION TITLE
Strange behavior I2C using invensense-imu library

### DISCUSSION THREAD

**Discussion by brightproject** [stm32duino/Arduino_Core_STM32#2062] [Libraries]
[Microcontroller](https://stm32-base.org/boards/STM32F401CCU6-WeAct-Black-Pill-V1.2) `stm32f401ccu6`
Connecting by I2c `SDA-PB7`, `SCL-PB6`
IMU `MPU-9250`
![71sAqWIEQwL _AC_SL1100_](https://github.com/stm32duino/Arduino_Core_STM32/assets/1788098/9773a3ae-9530-4f52-928a-934c4f06fdff)
This [sketch](https://github.com/stm32duino/Arduino_Core_STM32/blob/main/libraries/Wire/examples/i2c_scanner/i2c_scanner.ino)
Works good
 `I2C device found at address 0x68 !`
But when I try to use [sketch](https://github.com/bolderflight/invensense-imu/blob/main/examples/arduino/mpu9250/i2c/i2c.ino)
I get a message
`Error initializing communication with IMU`
Which says that the `wire` does not work.
I can't find the reason for this behaviour.
The author of the library [says](https://github.com/bolderflight/invensense-imu/issues/120) the reason is incorrect initialization of I2C.
Help me to understand.

**Comment by fpistm** [comment]
Hi @brightproject 
I convert your issue as a discussion as it concern a third party library.
The TwoWire implementation in this core follows the Arduino API. As I do not have this hardware I could not test but you have to check if PU resistor are available on both I2C line.
Then you can try to debug on your side to see why it failed, as I could not do that I could not help more. Maybe you can ask also on the forum.

**Comment by brightproject** [comment]
> The TwoWire implementation in this core follows the Arduino API. As I do not have this hardware I could not test but you have to check if PU resistor are available on both I2C line.

Thanks for the answer.
Currently, I am trying to find the root cause - the `STM32` core and `I2C` or a library for working with the `MPU9250` sensor
The [I2C](https://github.com/bolderflight/invensense-imu/issues/120#issuecomment-1628938140) bus works, but not always.
I also changed the pull-up resistors from `10 kΩ` to `4.7 kΩ`.
Additionally, I changed the address of the `MPU9250` sensors from `0x68` to `0x69`, by contact 3.3. volts at `AD0`.
The library still does not want to work with the sensor.
